### PR TITLE
Add Rubocop and Rspec tests for model

### PR DESCRIPTION
**Pull request by kasiaekiert**

### CODE REVIEW

File: spec/models/task_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Task, type: :model do
+  it 'is valid with name, content and deadline' do
+    task = Task.new(
```
Comment (koszany) [inline spec/models/task_spec.rb]: DRY with `let`

File: spec/models/task_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      name: 'Test task',
```
Comment (koszany) [inline spec/models/task_spec.rb]: add factoryBot

File: spec/models/task_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  it 'is invalid without a content' do
+    task = Task.new(content: nil)
+    task.valid?
+    expect(task.errors[:content]).to include("can't be blank")
+  end
+
+  it 'is invalid with a deadline in the past' do
+    task = Task.new(deadline: Date.today - 1.day)
```
Comment (koszany) [inline spec/models/task_spec.rb]:
```suggestion
    task = Task.new(deadline: Date.current - 1.day)
```

File: spec/models/task_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      name: 'Test task',
+      content: 'Test content',
+      deadline: Date.today + 1.day
+    )
+    expect(task).to be_valid
+  end
+
+  it 'is invalid without a name' do
+    task = Task.new(name: nil)
+    task.valid?
```
Comment (koszany) [inline spec/models/task_spec.rb]: add assertion for `#valid?`

File: spec/models/task_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    expect(task.errors[:name]).to include("can't be blank")
+  end
+
+  it 'is invalid without a content' do
```
Comment (koszany) [inline spec/models/task_spec.rb]:
```suggestion
  it "is invalid without a content" do
```

File: Gemfile [hunk 6]
```diff
@@ -1,36 +1,38 @@
...
+# frozen_string_literal: true
+
+source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby "3.0.1"
+ruby '3.0.1'
 gem 'bootstrap', '~> 5.1', '>= 5.1.3'
-
+gem 'rubocop', require: false
```
Comment (koszany) [inline Gemfile:8]: install `rubocop-rspec`

File: spec/models/task_spec.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Task, type: :model do
+  it 'is valid with name, content and deadline' do
+    task = Task.new(
+      name: 'Test task',
+      content: 'Test content',
+      deadline: Date.today + 1.day
```
Comment (koszany) [inline spec/models/task_spec.rb]: add example for `Date.current`

File: spec/models/task_spec.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    task = Task.new(content: nil)
+    task.valid?
+    expect(task.errors[:content]).to include("can't be blank")
+  end
+
+  it 'is invalid with a deadline in the past' do
+    task = Task.new(deadline: Date.today - 1.day)
+    task.valid?
+    expect(task.errors[:deadline]).to include("can't be in the past")
+  end
```
Comment (koszany) [inline spec/models/task_spec.rb:72]: add specs for `done`

File: spec/models/task_spec.rb (hunk 1 above)
Comment (koszany) [reply spec/models/task_spec.rb]: `let` vs `let!`

File: spec/models/task_spec.rb (hunk 4 above)
Comment (kasiaekiert) [reply spec/models/task_spec.rb]: what do you mean here?

Comment (koszany) [reply spec/models/task_spec.rb]: I mean check with `expect` if valid? returns true/false

File: spec/models/task_spec.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    expect(task.errors[:name]).to include("can't be blank")
+  end
+
+  it 'is invalid without a content' do
+    task = Task.new(content: nil)
+    task.valid?
+    expect(task.errors[:content]).to include("can't be blank")
+  end
+
+  it 'is invalid with a deadline in the past' do
```
Comment (koszany) [inline spec/models/task_spec.rb]: what if deadline is `Date.current`? What's the expected behaviour?

File: app/controllers/tasks_controller.rb [hunk 10]
```diff
@@ -1,31 +1,31 @@
+# frozen_string_literal: true
+
 class TasksController < ApplicationController
-  before_action :set_task, only: %i[ show edit update destroy ]
+  before_action :set_task, only: %i[show edit update destroy]
```
Comment (koszany) [inline app/controllers/tasks_controller.rb:4]: nice!

File: app/models/task.rb [hunk 11]
```diff
@@ -1,10 +1,12 @@
...
 class Task < ApplicationRecord
   validates_presence_of :name, :content
   validate :deadline_cannot_be_in_the_past
- 
+
   def deadline_cannot_be_in_the_past
-    if deadline.present? && deadline < Date.today
-      errors.add(:deadline, "can't be in the past")
-    end
+    return unless deadline.present? && deadline < Date.today
```
Comment (koszany) [inline app/models/task.rb:8]: nice guard clause 👍

File: spec/models/task_spec.rb (hunk 5 above)
Comment (kasiaekiert) [reply spec/models/task_spec.rb]: Can we leave it ' ' - for rubocop ?

File: Gemfile (hunk 6 above)
Comment (kasiaekiert) [reply Gemfile:8]: done

**Pull request merged**